src/test/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesStoreTest.kt

```diff
@@ -177,6 +181,199 @@ class ParticipantProjectSpeciesStoreTest : DatabaseTest(), RunsAsUser {
+      insertDeliverable(moduleId = moduleId, deliverableTypeId = DeliverableType.Species)
+
+      val participantId = insertParticipant(cohortId = cohortId)
+      val projectId1 = insertProject(participantId = participantId)
+      val projectId2 = insertProject(participantId = participantId)
+
+      val speciesId = insertSpecies()
+      val participantProjectSpeciesId1 =
+          insertParticipantProjectSpecies(projectId = projectId1, speciesId = speciesId)
+      val participantProjectSpeciesId2 =
+          insertParticipantProjectSpecies(projectId = projectId2, speciesId = speciesId)
+
+      every { user.canReadProjectDeliverables(any()) } returns false
+
+      assertEquals(
+          listOf(
+              ParticipantProjectsForSpecies(
+                  activeDeliverableId = null,
+                  participantProjectSpeciesId = participantProjectSpeciesId1,
+                  participantProjectSpeciesSubmissionStatus = SubmissionStatus.NotSubmitted,
+                  projectId = projectId1,
+                  projectName = "Project 1",
+                  speciesId = speciesId),
+              ParticipantProjectsForSpecies(
+                  activeDeliverableId = null,
+                  participantProjectSpeciesId = participantProjectSpeciesId2,
+                  participantProjectSpeciesSubmissionStatus = SubmissionStatus.NotSubmitted,
+                  projectId = projectId2,
+                  projectName = "Project 2",
+                  speciesId = speciesId)),
+          store.fetchParticipantProjectsForSpecies(inserted.organizationId, speciesId))
+    }
+
+    @Test
+    fun `returns an empty list of the user does not have permission to read the project`() {
+      val cohortId = insertCohort()
+      val moduleId = insertModule()
+      insertCohortModule(cohortId = cohortId, moduleId = moduleId)
+      insertDeliverable(moduleId = moduleId, deliverableTypeId = DeliverableType.Species)
+
+      val participantId = insertParticipant(cohortId = cohortId)
+      val projectId = insertProject(participantId = participantId)
+
+      val speciesId = insertSpecies()
+      insertParticipantProjectSpecies(projectId = projectId, speciesId = speciesId)
+
+      every { user.canReadProject(any()) } returns false
+
+      assertEquals(
+          emptyList<ParticipantProjectsForSpecies>(),
+          store.fetchParticipantProjectsForSpecies(inserted.organizationId, speciesId))
+    }
+  }
+
+  @Nested
+  inner class FetchSpeciesForParticipantProject {
+    @Test
+    fun `fetches the species and participant project species data associated to a participant project`() {
+      val cohortId = insertCohort()
+      val participantId = insertParticipant(cohortId = cohortId)
+      val projectId = insertProject(participantId = participantId)
+
+      val speciesId1 = insertSpecies(scientificName = "Acacia Kochi")
+      val speciesId2 = insertSpecies(scientificName = "Acacia Koa")
```

Comment:
Any reason this uses two very similar-looking species names?

Reply:
Not any good reason